File: mobile/src/main/java/com/ataulm/whatsnext/letterboxd/ApiFilmRelationship.java
package com.ataulm.whatsnext.letterboxd;

import com.google.gson.annotations.SerializedName;

class ApiFilmRelationship {

    @SerializedName("watched")
    boolean watched;

    @SerializedName("liked")
    boolean liked;

    @SerializedName("inWatchlist")
    boolean inWatchlist;

    @SerializedName("rating")
    double rating;
}

File: letterboxd/src/main/java/com/ataulm/letterboxd/ApiFilmRelationship.kt
package com.ataulm.letterboxd

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class ApiFilmRelationship(
    @field:Json(name = "watched")
    val watched: Boolean,

    @field:Json(name = "liked")
    val liked: Boolean,

    @field:Json(name = "inWatchlist")
    val inWatchlist: Boolean,

    @field:Json(name = "rating")
    val rating: Float?
)
